packages/contracts/contracts/ChainlinkAdapter.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {AggregatorV3Interface} from "./Dependencies/AggregatorV3Interface.sol";
+
+contract ChainlinkAdapter is AggregatorV3Interface {
+    uint8 public constant override decimals = 18;
+    uint256 public constant override version = 1;
+
+    /**
+     * @notice Maximum number of resulting and feed decimals
+     */
+    uint8 public constant MAX_DECIMALS = 18;
+
+    /// @notice PriceFeed always fetches current and previous rounds. It's ok to
+    /// hardcode round IDs as long as they are greater than 0.
+    uint80 public constant CURRENT_ROUND = 2;
+    uint80 public constant PREVIOUS_ROUND = 1;
+    int256 internal constant ADAPTER_PRECISION = int256(10 ** decimals);
+
+    /**
+     * @notice Price feed for (BTC / USD) pair
+     */
+    AggregatorV3Interface public immutable BTC_USD_CL_FEED;
+
+    /**
+     * @notice Price feed for (ETH / USD) pair
+     */
+    AggregatorV3Interface public immutable ETH_USD_CL_FEED;
+
+    int256 internal immutable BTC_USD_PRECISION;
+    int256 internal immutable ETH_USD_PRECISION;
+
+    constructor(AggregatorV3Interface _btcUsdClFeed, AggregatorV3Interface _ethUsdClFeed) {
+        BTC_USD_CL_FEED = AggregatorV3Interface(_btcUsdClFeed);
+        ETH_USD_CL_FEED = AggregatorV3Interface(_ethUsdClFeed);
+
+        require(BTC_USD_CL_FEED.decimals() <= MAX_DECIMALS);
+        require(ETH_USD_CL_FEED.decimals() <= MAX_DECIMALS);
+
+        BTC_USD_PRECISION = int256(10 ** BTC_USD_CL_FEED.decimals());
+        ETH_USD_PRECISION = int256(10 ** ETH_USD_CL_FEED.decimals());
+    }
+
+    function description() external view returns (string memory) {
+        return "BTC/ETH Chainlink Adapter";
+    }
+
+    function _min(uint256 _a, uint256 _b) private pure returns (uint256) {
+        return _a < _b ? _a : _b;
+    }
+
+    function _convertAnswer(int256 ethUsdPrice, int256 btcUsdPrice) private view returns (int256) {
+        return
+            (ethUsdPrice * BTC_USD_PRECISION * ADAPTER_PRECISION) /
```

Comment:
Q: Are you sure Precision is correct?

Is ADAPTER_PRECISION removed later? (doesn't seem like it)

Reply:
I think this converts the final price to adapter precision. For example, assuming ethUsdPrice = 8 and btcUsdPrice = 8, you'll get 8 + 8 + 18 - 8 - 8 = 18.

Reply:
Will double check the full path tomorrow when fresh, is there a E2E test for this?

Reply:
ok sounds good! This test uses real values from both BTC/USD and ETH/USD feeds. The output looks reasonable for a BTC/ETH feed.

https://github.com/ebtc-protocol/ebtc/pull/763/files#diff-257dca24d0b0a02c371b1e58901d2ac4bb520d5a1902aba611828faef676b892R19

I can plug the adapter into PriceFeed and double check.